### PR TITLE
[Aftershock] Add Habitat Block 1 to Exoplanet. Also adds Reinforced Glass Display cases and Electronic Lock doors.

### PR DESCRIPTION
**Pull request by QuillInkwell**
#### Summary
Content "Adds one variant of Habitat Block to the Aftershock Exoplanet. Also adds hackable display cases and doors."

#### Purpose of change
There is a need for more variety of structures in the Aftershock Exoplanet mod. This PR is second in a series of PRs aimed at increasing the number of locations on the Aftershock Exoplanet. This PR aims to tackle Habitat Blocks which are referred to at several points in the project repo but not currently implemented.

#### Describe the solution
This PR adds one variant of Habitat Block.
image_2024-05-29_144319618
The First floor is composed of a corner parking lot with two spaces. (I imagine not many people on Salus have cars since I never see them but I have no idea what the cannon is. I also couldn't fit more.) Right behind the parking lot is a wing of 8 motel style rooms. Compact seems to be an operative word for Salus IV. So for this Hab Block I tried to keep the resident spaces as minimal as possible. Each room is equipped with the bare minimum: 2 sleep pods, kitchenette and fridge, storage board, table and chair, and one tile washroom.

The front airlock leads into a lobby which connects the services to the residential wing.  In the lobby is a duo of vending machines selling food stuffs. There is also a nice bench and display of native alien flora. (Not sure if it's toxic, opps!) The two bathrooms connected to the lobby are guaranteed to spawn. As well as a the laundromat which comes equipped with ATM meant to imply a coin machine.

The other 3 rooms are all populated using nested map gen.
Slot 1 is located left of the airlock and will always be used by a dining related service. Currently it can be either a Cafe or a Dinner.
image
Slot 1 Cafe
image_2024-05-29_145507085
Slot 1 Diner

Slot 2 is located right of the Airlock and always filled a commercial service if some kind. Currently it can be:
Grocer 20%, Library 20%, Home Robotics Shop 10%, Civilian Gun Shop 10%, Clothes Shop 20%, or Pharmacy 10%
image_2024-05-29_145839076
Slot 2 Grocer

image
Slot 2 Gun Store (Front Door and Stock door are sealed by hackable metal doors. Guns are stored in hackable display cases.)

image_2024-05-29_150103285
Slot 2 Robotics Shop. (Front door and stock door are sealed by hackable metal doors. Robots are sealed in hackable display cases. Robots spawned are small mostly consumer/commercial grade bots. (See Domestic robot storage group))

image_2024-05-29_150321574
Slot 2 Clothes store

image_2024-05-29_150529415
Slot 2 Hardware Store

image_2024-05-29_150632623
Slot 2 Library

image_2024-05-29_150738689
Slot 2 Pharmacy

Slot 3 is located straight ahead of after the Airlock. Slot 3 is always filled with some resident service. Currently it can be:
Indoor Garden, Arcade, Sauna, Resident Storage Lockers, or Gym

image_2024-05-29_151059443
Slot 3 Resident Storage Lockers

image_2024-05-29_151143740
Slot 3 Sauna

image_2024-05-29_151215416
Slot 3 Arcade

image_2024-05-29_151258715
Slot 3 Gym

image_2024-05-29_151404561
Slot 3 Indoor Garden

And that was the first floor.
image_2024-05-29_151529481
Floor two contains another 8 motel style rooms located directly above the first wing. Fire escapes line the sides of the residential wing. At the end of the hall is a life support room which houses most of the buildings related life support systems. Beyond the airlock a large section of roof from the first floor is accessible for colonist use. The washrooms are guaranteed to spawn. Beyond the roof can be slotted with:
Xeno Garden, Community Garden, Aerocraft Landing Pad, Solar Panels, and mostly empty.

image
Xeno Garden

image_2024-05-29_152001521
Solar Panels

image
Community Garden

image
Aerocraft Pad (Some sealed civilian crates on the roof.)

image
Mostly Empty

#### Describe alternatives you've considered
I've considered using slightly more static Mapgen to save time. However I do like the feeling of smaller services clustered into larger structures. Since it allows colonists to access more services without having to exit climate controlled environments.

I've considered adding even more variants for each slot but I should probably leave this one alone for now.

I strongly considered making a a set of structures shells that are populated using nested mapgen. Structures that conditionally interlock with adjacent structures using tubes. However I had serious concerns over:
1. Complexity of implementation,
2. Lower quality of output in the event I am unable to adequately quality control the output of the procedural generation,
3. Balance concerns regarding basically combining multiple structures into one super structure. That can be navigated without ever going outside. 
4. The original design concept worked off single tile structures. However the proposed design felt cluttered with connective tubes. Which also obviously highlighted the grid like pattern of CDDA generation. If an implementation was to be pursued it would likely vastly prefer larger structures spanning multiple overmap terrains.

Ultimately I want to explore the concept more but am not ready to put into this structure.

#### Testing
Loaded into an Exoplanet world and verified the structures were spawning. Teleported to the structures to verify presence of enemies and correct execution of nested Mapgen. Verified all variants spawning. Eyeballed the loot amounts to decent(ish) values. (Feedback and balance adjustments always welcome).

### CODE REVIEW

**Comment by John-Candlebury** [comment]
Looks really nice!

**Comment by AltairreAtsu** [comment]
> Looks really nice!

Glad to hear.
I swear I had notepad++ search the entirety of my mod data multiple times to try and find that spelling mistake but evidently I did something wrong. At one point I just thought github was gaslighting me.

File: data/mods/Aftershock/maps/mapgen_pallete/afs_habitat_block.json [hunk 1]
```diff
@@ -0,0 +1,105 @@
+[
+  {
+    "type": "palette",
+    "id": "afs_habitat_block",
```
[the remark this answers is not in the document]
Comment (AltairreAtsu) [reply data/mods/Aftershock/maps/mapgen_pallete/afs_habitat_block.json:4]: Understood. I didn't want to step any toes and I knew I was going to be using a lot of furniture so I made my own palette. But I can try stick to using and expanding the existing ones.
I am still learning the dos and don'ts of mapgen and I can only hope I don't accidently create too much work for myself and others down the line.

[the remark this answers is not in the document]
Comment (John-Candlebury) [reply data/mods/Aftershock/maps/mapgen_pallete/afs_habitat_block.json:4]: Np! You are helping a bunch.
My advice is to use the palette to spawn low importance "background details" that can be shared by multiple buildings and explicitly list the main "loot" of the location in the map definition.  
So if for example you are making a gunstore, the guns and ammo in the racks can be mapped to symbols unique to the map piece, while you use the palette spawns for other background items like the cash register

**Review by John-Candlebury — APPROVED** [review]

**Pull request merged**